app/src/test/kotlin/model/WeightedDirectedGraphTest.kt

```diff
@@ -306,10 +307,97 @@ class WeightedDirectedGraphTest {
         }
     }
 
+    @Nested
+    inner class FindKeyVerticesTest {
+        @Nested
+        inner class `One vertex is picked over another`() {
+            @Test
+            fun `if it can reach more vertices`() {
+                val v0 = graph.addVertex(0)
+                val v1 = graph.addVertex(1)
+                val v2 = graph.addVertex(2)
+                val v3 = graph.addVertex(3)
+
+                graph.apply {
+                    addEdge(v0, v1, 1)
+                    addEdge(v0, v2, 1)
+                    addEdge(v0, v3, 1)
+                    addEdge(v1, v2, 1)
+                }
+
+                val expectedResult = setOf(v0)
+                val actualResult = graph.findKeyVertices()
+
+                assertEquals(expectedResult, actualResult)
+            }
+
+            @Test
+            fun `if it can reach other vertices with fewer edges`() {
+                val v0 = graph.addVertex(0)
+                val v1 = graph.addVertex(1)
+                val v2 = graph.addVertex(2)
+                val v3 = graph.addVertex(3)
+
+                graph.apply {
+                    addEdge(v0, v2, 1)
+                    addEdge(v0, v3, 1)
+                    addEdge(v1, v2, 1)
+                    addEdge(v2, v3, 1)
+                }
+
+                val expectedResult = setOf(v0)
+                val actualResult = graph.findKeyVertices()
+
+                assertEquals(expectedResult, actualResult)
+            }
+
+            @Test
+            fun `if its sum of distances to other vertices is less`() {
+                val v0 = graph.addVertex(0)
+                val v1 = graph.addVertex(1)
+                val v2 = graph.addVertex(2)
+                val v3 = graph.addVertex(3)
+
+                graph.apply {
+                    addEdge(v0, v2, 1)
+                    addEdge(v0, v3, 1)
+                    addEdge(v1, v2, 2)
+                    addEdge(v2, v3, 2)
+                }
+
+                val expectedResult = setOf(v0)
+                val actualResult = graph.findKeyVertices()
+
+                assertEquals(expectedResult, actualResult)
+            }
+        }
+
+        @Nested
+        inner class `Returns null`() {
+            @Test
+            fun `if graph has negative edges`() {
+                val v0 = graph.addVertex(0)
+                val v1 = graph.addVertex(1)
+                val v2 = graph.addVertex(2)
+
+                graph.apply {
+                    addEdge(v0, v1, 1)
+                    addEdge(v0, v2, -1)
+                }
+
+                val actualResult = graph.findKeyVertices()
+
+                assertNull(actualResult)
+            }
+        }
+    }
+
     @Nested
     inner class findShortestPathFordBellmanTest {
+
```

Comment:
@kar1mgh said, that we need to avoid placing new lines after class declaration( I think you need to refactor

Reply:
Done